packages/widget/src/utils/getSynapsePauseData.ts

```diff
@@ -0,0 +1,70 @@
+import { PAUSED_CHAINS_URL, PAUSED_MODULES_URL } from '@/constants/index'
+import { fetchJSONData } from '@/utils/fetchJsonData'
+
+enum SessionStorageKey {
+  CHAIN_PAUSE = 'synapse-paused-chains',
+  MODULE_PAUSE = 'synapse-paused-modules',
+  TIMESTAMP = 'synapse-paused-data-timestamp',
+}
+
+export const getSynapsePauseData = () => {
+  const fetchAndStoreData = async () => {
+    try {
+      console.log('[Synapse Widget] Fetching pause data')
+      const chainsData = await fetchJSONData(PAUSED_CHAINS_URL)
+      const modulesData = await fetchJSONData(PAUSED_MODULES_URL)
+
+      sessionStorage.setItem(
+        SessionStorageKey.CHAIN_PAUSE,
+        JSON.stringify(chainsData)
+      )
+      sessionStorage.setItem(
+        SessionStorageKey.MODULE_PAUSE,
+        JSON.stringify(modulesData)
+      )
+      sessionStorage.setItem(SessionStorageKey.TIMESTAMP, Date.now().toString())
+    } catch (error) {
+      console.error(
+        '[Synapse Widget] Failed to fetch paused chains/modules: ',
+        error
+      )
+    }
+  }
+
+  const readData = () => {
+    if (sessionStorage) {
+      const pausedChainsData = JSON.parse(
+        sessionStorage.getItem(SessionStorageKey.CHAIN_PAUSE)
+      )
+      const pausedModulesData = JSON.parse(
+        sessionStorage.getItem(SessionStorageKey.MODULE_PAUSE)
+      )
+      return { pausedChainsData, pausedModulesData }
+    }
+    return { pausedChainsData: null, pausedModulesData: null }
+  }
+
+  const checkIsDataValid = (): boolean => {
+    const lastFetchTime = sessionStorage.getItem(SessionStorageKey.TIMESTAMP)
+
+    if (lastFetchTime) {
+      const previousTime = Number(lastFetchTime)
+      const currentTime = Date.now()
+
+      const msPerHr = 1000 * 60 * 60
+      const timeElapsedInHrs = (currentTime - previousTime) / msPerHr
+
+      return timeElapsedInHrs < 1
+    } else {
+      return false
+    }
+  }
+
+  const isValid = checkIsDataValid()
```

Comment:
Is this so it only updates the session store after an hour has passed? If it's to prevent http calls, don't think it's worth it since there's a condition where a user used the widget in the time between last use and when we push a chain pause. Fine to check each time.

Reply:
Thanks, updated to remove `checkIsDataValid()`, now checking every time in https://github.com/synapsecns/sanguine/commit/fc8c79c22d2be6c8d2555359fdb877e777a675aa

Added `isFetching` flag to prevent duplicate fetches on re-renders.